src/renderer/reducers/settings.js

```diff
@@ -92,6 +92,7 @@ export type SettingsState = {
   hideEmptyTokenAccounts: boolean,
   sidebarCollapsed: boolean,
   discreetMode: boolean,
+  OSDarkMode: boolean,
```

Comment:
I'm not sure this should be in the settings because it seems you want to make this state reflect the exact state of the UI. 
we are trying to catch up the user settings from the user system context, which is a bit unoptimized (two source of state – and basically it's not something we should/need to save into the app.json / settings). 
even tho I understand the challenge in term of re-rendering I think either we make a new reducers (that is not saved in app.json) to reflect the "window context" or we find another way?

Reply:
Moved the application state as discussed, without the initial dispatch